src/os/bluestore/BlueFS.h

```diff
@@ -443,7 +443,7 @@ class BlueFS {
 
     void append(uint64_t value) {
       uint64_t l0 = get_buffer_length();
-      ceph_assert(l0 + sizeof(value) <= std::numeric_limits<unsigned>::max());
+      ceph_assert(l0 + sizeof(value) <= std::numeric_limits<uint64_t>::max());
```

Comment:
I think we actually want check that we do not overflow 32 bits:
`ceph_assert(l0 + sizeof(value) <= (uint64_t)std::numeric_limits<uint32_t>::max());`